docs/Streams/index.md

```diff
@@ -0,0 +1,83 @@
+### Streams
+
+There are currently two Streams avaliable, `CF_BinaryStream` and `CF_StringStream`.
+
+`CF_BinaryStream` is optimized for working with binary files and allowing reading of characters, bytes, ints, floats, booleans and vectors.
+
+`CF_StringStream` is optimized for working with text files and allows reading of characters.
+
+### Working with files
+
+Using the `File` method in the target stream, different `FileMode`'s can be used. 
+
+**text.txt**:
+```
+Hello, World!
+```
+
+#### Read
+
+```
+CF_StringStream stream();
+stream.File("$profile:test.txt", FileMode.READ);
```

Comment:
Maybe have a look at c++ fstream. It had an optional constructor parameter where you could put in the file and the mode already which internally called Open

I think binarystream and stringstream are ok, but if you want to interact with files, then that should be a CF_Filestream ... which you could open from one of the other streams to get their data into the file or read from the file again. E.g.

```cs
CF_StringStream data();
data.Write("Hello Jacob");
CF_FileStream fileWrite("<path>", Filemode.OPTION);
fileWrite.Write(data);
fileWrite.Close();
```

Your examples all show file interaction ... maybe you should first demonstrate the streams for what they are named after, aka reading and writing strings or binary data ... and the use-case together with files should be coming a bit later - or be its own kind of stream.